src/backend/distributed/commands/call.c

```diff
@@ -101,51 +104,33 @@ CallFuncExprRemotely(CallStmt *callStmt, DistObjectCacheEntry *procedure,
 
 	CitusTableCacheEntry *distTable = GetCitusTableCacheEntry(colocatedRelationId);
 	Var *partitionColumn = distTable->partitionColumn;
+	bool colocationWithReferenceTable = false;
 	if (partitionColumn == NULL)
 	{
 		/* This can happen if colocated with a reference table. Punt for now. */
 		ereport(DEBUG1, (errmsg(
-							 "cannot push down CALL for reference tables")));
-		return false;
+							 "will push down CALL for reference tables")));
+		colocationWithReferenceTable = true;
+		Assert(IsReferenceTable(ColocatedTableId));
```

Comment:
`ColocatedTableId` is a function, this probably won't end well.

Reply:
I meant to have `colocatedRelationId` here. I did not see the compiler warnings, I guess assertions are not enabled on my builds. I need to fix this as well.